contracts/stake/src/contract.rs

```diff
@@ -631,8 +641,85 @@ impl StakingTrait for Staking {
             .publish(("fund_distribution", "end_time"), end_time);
     }
 
+    fn update_config(
+        env: Env,
+        lp_token: Option<Address>,
+        min_bond: Option<i128>,
+        min_reward: Option<i128>,
+        manager: Option<Address>,
+        owner: Option<Address>,
+        max_complexity: Option<u32>,
+    ) -> Result<Config, ContractError> {
+        env.storage()
+            .instance()
+            .extend_ttl(INSTANCE_RENEWAL_THRESHOLD, INSTANCE_TARGET_TTL);
+
+        let admin = get_admin_old(&env);
+        admin.require_auth();
+
+        let mut config = get_config(&env);
+
+        if let Some(lp_token) = lp_token {
+            config.lp_token = lp_token;
+        }
+
+        if let Some(min_bond) = min_bond {
+            if min_bond <= 0 {
+                log!(
+                &env,
+                "Stake: initialize: Minimum amount of lp share tokens to bond can not be smaller or equal to 0"
+            );
+                panic_with_error!(&env, ContractError::InvalidMinBond);
+            }
+            config.min_bond = min_bond
+        }
+
+        if let Some(min_reward) = min_reward {
+            if min_reward <= 0 {
+                log!(&env, "Stake: initialize: min_reward must be bigger than 0!");
+                panic_with_error!(&env, ContractError::InvalidMinReward);
+            }
+            config.min_reward = min_reward
+        }
+
+        if let Some(manager) = manager {
+            config.manager = manager;
+        }
+
+        if let Some(owner) = owner {
+            config.owner = owner;
+        }
+
+        if let Some(max_complexity) = max_complexity {
+            if max_complexity == 0 {
+                log!(
+                    &env,
+                    "Stake: initialize: max_complexity must be bigger than 0!"
+                );
+                panic_with_error!(&env, ContractError::InvalidMaxComplexity);
+            }
+            config.max_complexity = max_complexity
+        }
+
+        save_config(&env, config.clone());
+
+        Ok(config)
+    }
+
     // QUERIES
 
+    // from an old tag version 1.1.0
+    // fn query_distributed_rewards(env: Env, asset: Address) -> u128 {
+    //     let staking_rewards = find_stake_rewards_by_asset(&env, &asset).unwrap();
+    //     let unds_rew_fn_arg: Val = asset.into_val(&env);
+    //     let ret: u128 = env.invoke_contract(
+    //         &staking_rewards,
+    //         &Symbol::new(&env, "query_distributed_reward"),
+    //         vec![&env, unds_rew_fn_arg],
+    //     );
+    //     ret
+    // }
```

Comment:
Okay but why adding it here?

Reply:
thought it'd be needed in later iteration. We had it commented out for a long time in the old/tagged version. Removed now